diff -dur -x '*~' -x '*.orig' -x '*.rej' openjdk.orig/common/autoconf/platform.m4 openjdk/common/autoconf/platform.m4
--- openjdk.orig/common/autoconf/platform.m4	2016-07-25 07:38:20.000000000 +0200
+++ openjdk/common/autoconf/platform.m4	2016-09-20 09:21:06.000000000 +0200
@@ -36,6 +36,12 @@
       VAR_CPU_BITS=64
       VAR_CPU_ENDIAN=little
       ;;
+    x32)
+      VAR_CPU=x86_64
+      VAR_CPU_ARCH=x86
+      VAR_CPU_BITS=32
+      VAR_CPU_ENDIAN=little
+      ;;
     i?86)
       VAR_CPU=x86
       VAR_CPU_ARCH=x86
diff -dur -x '*~' -x '*.orig' -x '*.rej' openjdk.orig/hotspot/src/os/linux/vm/os_linux.cpp openjdk/hotspot/src/os/linux/vm/os_linux.cpp
--- openjdk.orig/hotspot/src/os/linux/vm/os_linux.cpp	2016-09-20 09:17:58.000000000 +0200
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2016-09-20 09:19:16.000000000 +0200
@@ -1911,7 +1911,7 @@
 
   #if  (defined IA32)
     static  Elf32_Half running_arch_code=EM_386;
-  #elif   (defined AMD64)
+  #elif  (defined AMD64) || (defined X32)
     static  Elf32_Half running_arch_code=EM_X86_64;
   #elif  (defined IA64)
     static  Elf32_Half running_arch_code=EM_IA_64;
diff -dur -x '*~' -x '*.orig' -x '*.rej' openjdk.orig/hotspot/src/share/vm/utilities/taskqueue.hpp openjdk/hotspot/src/share/vm/utilities/taskqueue.hpp
--- openjdk.orig/hotspot/src/share/vm/utilities/taskqueue.hpp	2016-07-25 07:38:24.000000000 +0200
+++ openjdk/hotspot/src/share/vm/utilities/taskqueue.hpp	2016-09-20 09:22:34.000000000 +0200
@@ -400,7 +400,7 @@
   // Architectures with weak memory model require a barrier here
   // to guarantee that bottom is not older than age,
   // which is crucial for the correctness of the algorithm.
-#if !(defined SPARC || defined IA32 || defined AMD64)
+#if !(defined SPARC || defined IA32 || defined AMD64 || defined X32)
   OrderAccess::fence();
 #endif
   uint localBot = OrderAccess::load_acquire((volatile juint*)&_bottom);
diff -dur -x '*~' -x '*.orig' -x '*.rej' openjdk.orig/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c openjdk/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c
--- openjdk.orig/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c	2016-07-25 07:38:26.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c	2016-09-20 09:19:37.000000000 +0200
@@ -41,7 +41,6 @@
 #endif
 #ifdef __linux__
 #include <unistd.h>
-#include <sys/sysctl.h>
 #include <sys/utsname.h>
 #include <netinet/ip.h>
 
diff -dur -x '*~' -x '*.orig' -x '*.rej' openjdk.orig/jdk/src/solaris/native/java/net/PlainSocketImpl.c openjdk/jdk/src/solaris/native/java/net/PlainSocketImpl.c
--- openjdk.orig/jdk/src/solaris/native/java/net/PlainSocketImpl.c	2016-07-25 07:38:26.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/PlainSocketImpl.c	2016-09-20 09:19:37.000000000 +0200
@@ -43,7 +43,6 @@
 #endif
 #ifdef __linux__
 #include <unistd.h>
-#include <sys/sysctl.h>
 #endif
 
 #include "jvm.h"
