--- openjdk/hotspot/src/os/linux/vm/os_linux.cpp~	2015-12-24 07:19:42.000000000 +0100
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2019-11-18 20:11:28.343551301 +0100
@@ -370,9 +370,9 @@ void os::init_system_properties_values()
 //        ...
 //        7: The default directories, normally /lib and /usr/lib.
 #if defined(AMD64) || defined(_LP64) && (defined(SPARC) || defined(PPC) || defined(S390))
-#define DEFAULT_LIBPATH "/usr/lib64:/lib64:/lib:/usr/lib"
+#define DEFAULT_LIBPATH "/usr/lib64/java:/usr/lib64:/lib64:/lib:/usr/lib"
 #else
-#define DEFAULT_LIBPATH "/lib:/usr/lib"
+#define DEFAULT_LIBPATH "/usr/lib/java:/lib:/usr/lib"
 #endif
 
 // Base path of extensions installed on the system.
